app/root.tsx

```diff
@@ -25,6 +26,69 @@ import iconsHref from "~/icons.svg";
 import cx from "clsx";
 import { canUseDOM } from "./ui/primitives/utils";
 import { GlobalLoading } from "./ui/global-loading";
+import { Route } from "./+types/root";
+
+/**
+ * In preparation for using `remix` repo's `main` branch for v3 development,
+ * we've deployed the latest v2 docs at https://v2.remix.run/docs .
+ * Docs for specific versions of v0, v1, and v2 can be found on the `remix` repo within the `docs/` folder at the corresponding GitHub tag for that version.
+ *
+ * This middleware redirects requests for specific v0,v1, or v2 versions of the docs to GitHub
+ * and redirects requests for `main` and `dev` to the new static v2 docs site (https://v2.remix.run/docs).
+ */
+const v2DocsMiddleware: Route.unstable_MiddlewareFunction = ({ request }) => {
+  // https://github.com/remix-run/remix/tree/remix%402.17.0/docs
+  const { pathname } = new URL(request.url);
+
+  const segments = pathname.split("/").slice(1);
+  if (segments[0] !== "docs") return;
+
+  const [lang, ref, ...path] = segments.slice(1);
+  const doc = path.length > 0 ? path.join("/") + ".md" : "";
+
+  if (lang === "en") {
+    // /docs/en/v0.x -> github
+    // /docs/en/v1.x -> github
+    if (ref.startsWith("v0.") || ref.startsWith("v1.")) {
+      // We switched from `vx.y.z` tags to `remix@x.y.z` after v1.6.4 release
+      // So here we use leading `v` in docs ref to detect that older tag format
+      //
+      // See:
+      // - https://remix.run/docs/en/v1.6.4
+      // - https://remix.run/docs/en/1.6.5
+      //
+      // - https://github.com/remix-run/remix/releases/tag/v1.6.4
+      // - https://github.com/remix-run/remix/releases/tag/remix%401.6.5
+
+      // todo: remove leading `v` from newer versions like v1.6.5??
```

Comment:
For version 1.6.5 and newer, the server redirects things like `vx.y.z` to `x.y.z`. But that should only matter if you manually type that version number into the URL since nothing actually links to `v`-prefixed versions newer than 1.6.5. I assume we just added that at the time to smooth over the 1.6.5 release, but feels like its a non-issue at this point. But let me know what you think!

Reply:
Yeah, I've seen this before in [the react-router-website repo](https://github.com/remix-run/react-router-website/blob/main/app/modules/gh-docs/.server/tags.ts#L87-L91). I bet it's the same reason: with changesets we switched the format of the tags

Reply:
Oh yea I've been in the Remix v3 headspace where we are avoiding dependencies where ever possible, but forgot that this repo already has `semver` installed. I'll switch over to that